### ISSUE TITLE
BeeAgent with granite LLM uses BeeAssistantPrompt vs. GraniteBeeAssistantPrompt

### ISSUE THREAD

**Issue by michael-desmond** [i-am-bee/bee-agent-framework#219]
**Describe the bug**
I was noticing that BeeAgent when configured with a granite llm uses the standard BeeAssistantPrompt which conflicts with the granite system prompt and causes the llm to perform poorly. The offending code is here: 

https://github.com/i-am-bee/bee-agent-framework/blob/fd942ae0116893b9452530c305e0053bb83f5578/src/agents/bee/agent.ts#L133

A fairly simple fix is to conditionally choose a default AssistantPrompt before rendering i.e. 

```
const template = (this.input.llm.modelId.includes("granite"))?GraniteBeeAssistantPrompt: BeeAssistantPrompt;
```

But this is not the most elegant solution imo because it introduces more ad hoc checks into the agent. 

Ideally we could move the rendering of the assistant message into the runner hierarchy. The renders method seems to be the most appropriate spot. 

https://github.com/i-am-bee/bee-agent-framework/blob/fd942ae0116893b9452530c305e0053bb83f5578/src/agents/bee/runners/default/runner.ts#L256C17-L256C28

However the renders method is introduced on the DefaultRunner as opposed to the BaseRunner which is the type maintained by the BeeAgent. 

**To Reproduce**

1. Create a BeeAgent with granite and breakpoint line 130.
3. The BeeAssistantPrompt is applied when rendering the assistant prompt.

**Expected behavior**
The GraniteBeeAssistantPrompt should be applied when rendering the assistant prompt.

**Set-up:**

- Bee version: 0.0.47
- Model provider: Ollama

**Comment by michael-desmond** [comment]
@Tomas2D Can u give an opinion on this one when you get a chance.

**Comment by Tomas2D** [comment]
Good catch! 

Fixed in #223

**Comment by Tomas2D** [comment]
Released in v0.0.48
